### PR TITLE
Introduce ScanScope, a more flexible way to define what to scan.

### LINKED ISSUE #58: Drop SlotTable.scan, add composable selection support
Another solution to this problem, which would be much more useful for devs, would be the ability to start the scan at some point further down in the tree, similar to passing a non-root View to scan(). However that presents a few (solvable) issues:

1. Composables don't have any implicit identity, so we'd probably need to do something like for filtering and use test tags or layout IDs to identify the root.
2. The visitor infra doesn't currently support filtering nodes above, only below, so that would have to be implemented. This could also make it possible to visit Groups directly instead of using the intermediate LayoutInfo type - I'm not sure we _should_, since that type lets us decouple a bit from compose APIs that might change a lot, but we'd have the option. 

Having thought about this more, that API makes much more sense than this one. I'll take a stab at implementing it, but would like to not block this PR on it since this API is marked as unstable anyway.

_Originally posted by @zach-klippenstein in https://github.com/square/radiography/pull/33#discussion_r474794986_

### PR DESCRIPTION
**Pull request by zach-klippenstein**
Based on #69.
Closes #58.

`FocusedWindowViewFilter` is now `FocusedWindowScope`. `SlotTable.scan()` is gone.

### CODE REVIEW

File: radiography/src/main/java/radiography/Radiography.kt [hunk 1]
```diff
@@ -40,69 +38,70 @@ public object Radiography {
   @JvmStatic
   @JvmOverloads
   public fun scan(
-    rootView: View? = null,
+    scanScope: ScanScope = AllWindowsScope,
```
Comment (pyricau) [inline radiography/src/main/java/radiography/Radiography.kt:41]: Makes me happy!

File: radiography/src/main/java/radiography/ScanScopes.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  /**
+   * Scans all Composables in all windows that have the `Modifier.testTag` modifier with the given
+   * [testTag].
+   */
+  @ExperimentalRadiographyComposeApi
+  @JvmStatic
+  @JvmOverloads
+  fun composeTestTagScope(
```
Comment (pyricau) [inline radiography/src/main/java/radiography/ScanScopes.kt:44]: Might want to explain what a TestTag is and even link to the doc. https://developer.android.com/reference/kotlin/androidx/compose/ui/platform/package-summary#(androidx.compose.ui.Modifier).testTag(kotlin.String)

> Applies a tag to allow modified element to be found in tests.

It's not clear how this relates to Radiography.

Comment (pyricau) [reply radiography/src/main/java/radiography/ScanScopes.kt:44]: filed https://github.com/square/radiography/issues/72

Comment (zach-klippenstein) [reply radiography/src/main/java/radiography/ScanScopes.kt:44]: Good catch, I linked to the docs in the `ViewFilter` for test tags, but forgot to do so here.

**Pull request merged**